documentation/docs/agent-configuration/config-agent-file.md

```diff
@@ -0,0 +1,116 @@
+# Config Agent File
+This guide explains the structure and options available within the agent config file. Use cases illustrating how to customise the config file for different scenarios are included. 
+## Diagram
+The diagram belows describes the structure of an agent file
+
+![agent file diagram](/img/agent-file.svg)
+
+## Constants
+This section is to declare all the constants of the agent.
+
+```yml
+constants:
+  baseUrl: <BASE URL>
+  port: <CUSTOM PORT>
+  dbEncryptionKey: <SECRET KEY>
+  databaseFile: <DATA BASE FILE>
+  methods:
+  - <PLUGIN FUNCTIONS>
+```
+* `baseUrl`: This specifies the base URL for the application or service. 
+* `port`: This specifies the port number on which the application will listen for incoming connections. 
+* `dbEncryptionKey`: This field contains a cryptographic key used for encrypting data in the database. It's a hexadecimal string representing the encryption key. Run this command `veramo config gen-key` to generate a key.
+* `databaseFile`: This specifies the path to the SQLite database file.
```

Comment:
Could I use other databases such as PostgreSQL or MongoDB? If I can, how can I configure it?

Reply:
you can use whatever type of database as long as it's supported by typeorm datasoure. I've updated the document to mention this.